src/python/pants/jvm/resolve/coursier_fetch.py

```diff
@@ -396,15 +397,15 @@ async def prepare_coursier_resolve_info(
 
     coord_arg_strings = set()
     force_version_coord_arg_strings = set()
-    for req in to_resolve:
+    for req in sorted(to_resolve, key=lambda ar: ar.coordinate):
```

Comment:
This is a drive by looking for Pex related issues / work, so take it with a grain of salt, but:

Great, sorted now ... but these req are not the inputs to `CoursierResolveInfo`. The inputs are `coord_arg_strings` and `force_version_coord_arg_strings` and those are sets still.

Is this a facepalm waiting to happen?

Reply:
You're absolutely correct, thanks for catching that oversight of mine; I really appreciate it.

I corrected my mistake by using an `OrderedSet` for both `coord_arg_strings` and `force_version_coord_arg_strings` which will preserve the order when turned into a `FrozenOrderedSet`.

I force-pushed this fix (with an additional rebase on `main`).